### PR TITLE
Show build outputs and unit test results in CircleCI Web UI

### PR DESCRIPTION
**Pull request by jamesonwilliams**
### Now, there is content in the CircleCI run's ["Tests"  tab](https://app.circleci.com/pipelines/github/aws-amplify/amplify-android/1761/workflows/86718868-f4d4-4295-b2ef-67e61c552796/jobs/3371/tests):
Screen Shot 2020-05-26 at 9 51 12 PM

### Now, there is content in the CircleCI ["Artifacts" Tab](https://app.circleci.com/pipelines/github/aws-amplify/amplify-android/1761/workflows/86718868-f4d4-4295-b2ef-67e61c552796/jobs/3371/artifacts):
Screen Shot 2020-05-26 at 9 51 36 PM

------------------------
Currently we are not able to see the outputs of unit tests that run in CircleCI.

To enable this, we need to package test results into a fake directory structure that CircleCI expects.

------------------------

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/aggregate-build-artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+           echo "$item"
+       fi
+   done
+}
+readonly modules=($(modules_with_build_trees))
+readonly current_dir="$(pwd)"
+readonly aggregation_dir="build-artifacts"
+
+if [ ! -d "$aggregation_dir" ]; then
+    mkdir "$aggregation_dir"
```
Comment (palpatim) [inline .circleci/aggregate-build-artifacts.sh:17]: `mkdir -p` would be slightly more robust in that it a) will make the full tree if the spec happens to need it, and b) won't fail if the directory suddenly starts to exist between the test and the `mkdir` command.

File: .circleci/aggregate-test-results.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    for item in *; do
+        if [ -d "$item/$junit_xmls_path" ]; then
+            echo $item
+        fi
+    done
+}
+readonly test_results_dirs=($(module_test_results))
+
+if [ ! -d $circle_ci_storage_dir ]; then
+    mkdir $circle_ci_storage_dir
```
Comment (palpatim) [inline .circleci/aggregate-test-results.sh:21]: As above, maybe `mkdir -p`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -90,12 +90,14 @@ jobs:
       - generate_gradle_wrapper
       - run:
           name: Build the project
-          command: bash gradlew build
-      - run:
-          name: Publish the project to local maven
-          command: bash gradlew publishToMavenLocal
```
Comment (palpatim) [inline .circleci/config.yml:96]: I assume the aggregate-build-artifacts takes the place of publishing to local maven?

Comment (jamesonwilliams) [reply .circleci/config.yml:96]: This `publishToMavenLocal` step is actually not needed. It was carried over into this project from the SDK -- there, it is important.

File: .circleci/aggregate-test-results.sh (hunk 2 above)
Comment (jamesonwilliams) [reply .circleci/aggregate-test-results.sh:21]: will move to this

File: .circleci/aggregate-build-artifacts.sh (hunk 1 above)
Comment (jamesonwilliams) [reply .circleci/aggregate-build-artifacts.sh:17]: will move to this

**Pull request merged**